spec/support/acceptance/tar.rb

```diff
@@ -0,0 +1,37 @@
+begin
+  require 'serverspec'
+rescue LoadError
+  # Not using acceptance tests
+else
+  module Serverspec
+    module Type
+      class Tar < Command
+        def contents
+          command_result.stdout
```

Comment:
Why do you treat this as a string? Shouldn't you split it into multiple lines? Then you can use `its(:contents) { should include('ssl-build/foreman-proxy.example.com/foreman-proxy.example.com-qpid-router-server.key') }`.

Reply:
The reason I suggest it is that you can then use `match_array`. https://relishapp.com/rspec/rspec-expectations/v/3-10/docs/built-in-matchers. Something like:
```ruby
let(:expected_files_in_tar) do
  [
    ...
  ]
end

describe tar('/path') do
   its(:contents) { should match_array(expected_files_in_tar) }
end
```

That also removes the need for `contents_count` since it will check the length. It will also show you which additional entries are there. That is something you don't have today.

Reply:
Ahh ha, excellent way to solve my problem. Thanks. I have implemented this and added a direct exist check.